tv_apps/smart-home/js/message_handler.js

```diff
@@ -37,12 +40,38 @@
     },
 
     handleActivity: function mh_handleActivity(activity) {
+      if(this.isStop) {
+        this.activityQueue.push(activity);
+        return;
+      }
+      this._digestActivity(activity);
+    },
+
+    stopActivity: function mh_stopActivity() {
+      this.isStop = true;
+    },
+
+    _digestActivity: function(activity) {
       var name = activity.source.name;
       switch(name) {
         case 'pin':
           this.pin(activity.source.data);
           break;
       }
+    },
+
+    digest: function mh_digest() {
```

Comment:
would it be better to change name to `resumeActivity`? since you have state changing inside and this is paired with `stopActivity`.